### DISCUSSION TITLE
How do I intercept event events and authenticate them in a unified manner?

### DISCUSSION THREAD

**Discussion by zhengdechang** [miguelgrinberg/Flask-SocketIO#2052] [Q&A]
How do I intercept event events and authenticate them in a unified manner? For example, flask processes before_request of http. Which socketio has a similar function

**Comment by miguelgrinberg** [comment]
Inidividual events do not need to be authenticated, because the Socket.IO connection is long lasting. You only need to authenticate in the connect event handler. Save any authentication state that you need there so that you can access it on subsequent events.

**Comment by zhengdechang** [comment]
I want to implement unified interception authentication and write the authentication for each event. It is likely that this is too many. Before obtaining a cookie, if it does not expire, the cookie is ignored

**Reply by miguelgrinberg** [reply]
As I said above, there is no need to authenticate each event. Socket.IO events are not individually sent like requests, they are all sent within the same long-lived connection.

**Comment by zhengdechang** [comment]
What if I want to refresh the cookie every time I communicate? Is there a uniform way

**Reply by miguelgrinberg** [reply]
You can't refresh cookies with Socket.IO. You are thinking in terms of HTTP requests. Socket.IO implements a long lived connection, if you need to send an update to the client, you cannot set a cookie, you just send an event with the updated data and it is up to the client to deal with this update in a way that makes sense.
